rtloader/common/builtins/datadog_agent.c

```diff
@@ -381,39 +386,45 @@ static PyObject *set_external_tags(PyObject *self, PyObject *args)
         // list must contain only tuples in form ('hostname', {'source_type': ['tag1', 'tag2']},)
         if (!PyTuple_Check(tuple)) {
             PyErr_SetString(PyExc_TypeError, "external host tags list must contain only tuples");
+            error = 1;
             goto error;
         }
 
         // first elem is the hostname
         hostname = as_string(PyTuple_GetItem(tuple, 0));
         if (hostname == NULL) {
             PyErr_SetString(PyExc_TypeError, "hostname is not a valid string");
+            error = 1;
             goto error;
         }
 
         // second is a dictionary
         PyObject *dict = PyTuple_GetItem(tuple, 1);
         if (!PyDict_Check(dict)) {
             PyErr_SetString(PyExc_TypeError, "second elem of the host tags tuple must be a dict");
+            error = 1;
             goto error;
         }
 
         // dict contains only 1 key, if dict is empty don't do anything
         Py_ssize_t pos = 0;
         PyObject *key = NULL, *value = NULL;
         if (!PyDict_Next(dict, &pos, &key, &value)) {
+            error = 1;
```

Comment:
we must not set error here since we want to ignore this item and simply continue iterating through the list